packages/core/src/util/gatewayRateLimitError.ts

```diff
@@ -0,0 +1,31 @@
+import type { GatewayOpcodeRateLimitMetadataMap, GatewayRateLimitedDispatchData } from 'discord-api-types/v10';
+
+/**
+ * Represents the error thrown when the gateway emits a RATE_LIMITED event after a certain request.
+ */
+export class GatewayRateLimitError extends Error {
+	public override readonly name = GatewayRateLimitError.name;
```

Comment:
I don't think this override is necessary?

<img width="303" height="74" alt="Image" src="https://github.com/user-attachments/assets/b3a4cd7e-510b-4d41-babb-89ffe9592109" />